src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryMath.cs

```diff
@@ -167,9 +167,16 @@ private class MinDateAndDateTimeAgg : IAggregator
         {
             protected DateTime _minValueDT = DateTime.MaxValue;
             protected int _count = 0;
+            private readonly IServiceProvider _serviceProvider;
+
+            public MinDateAndDateTimeAgg(IServiceProvider serviceProvider)
+            {
+                _serviceProvider = serviceProvider;
+            }
 
             public void Apply(FormulaValue value)
             {
+                var timeZoneInfo = Utility.GetService<TimeZoneInfo>(_serviceProvider);
```

Comment:
> var timeZoneInfo = Utility.GetService<TimeZoneInfo>(_serviceProvider); [](http://example.com/codeflow?start=15&length=71)

Apply() is a hot path. 

Just do this once in the ctor. 
In fact... you could chain ctors and just save the tzi:

            public MinDateAndDateTimeAgg(IServiceProvider serviceProvider) 
  : this(serviceProvider.GetService<TimeZoneInfo>)
            {
            }

            public MinDateAndDateTimeAgg(TZI)
            {
                _tzi = tzi;
            }

Reply:
updated.